mypy/checker.py

```diff
@@ -2067,6 +2067,9 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
                 and self.scope.active_class() is not None):
             self.fail(message_registry.DEPENDENT_FINAL_IN_CLASS_BODY, s)
 
+        if s.was_annotated and not self.in_checked_function():
```

Comment:
Is it possible to just use `s.unanalyzed_type is not None` instead of adding the new attribute?